src/applications/static-pages/facilities/vet-center/NearByVALocations.jsx

```diff
@@ -0,0 +1,235 @@
+          multiTypeQuery(
+            'Cemetery',
+            genQuery(boundingBox, coordinates, 'cemetery', false),
+          ),
+        );
+        dispatch(
+          multiTypeQuery(
+            'VetCenter',
+            genQuery(boundingBox, coordinates, 'vet_center', true),
+          ),
+        );
+      }
+    },
+    [props, dispatch],
+  );
+
+  useEffect(
+    () => {
+      if (!props.togglesLoading && !isStartedLoading(props.multiLoading)) {
+        fetchNearbyVALocations();
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [props.togglesLoading, props.multiLoading],
+  );
+
+  useEffect(
+    () => {
+      const noDistancesToMeasure =
+        originalCoordinates.length === 0 || !isFinishedLoading(props);
+
+      if (nearbyVADistances || noDistancesToMeasure) {
+        return false;
+      }
+
+      const facilityCoordinates = joinMultiData(props)
+        .filter(center => center.id !== props.mainFacilityApiId)
+        .map(center => ({
+          id: center.id,
+          coordinates: [center.attributes.long, center.attributes.lat],
+        }));
+
+      const fetchDrivingData = async () => {
+        if (nearbyVADistances) {
+          return;
+        }
+        const response = await fetch(
+          distancesToNearbyVetCenters(
+            originalCoordinates,
+            facilityCoordinates.map(center => center.coordinates),
+          ),
+        );
+        const data = await response.json();
+        const nearbyDistances = data.distances.map(distance =>
+          convertMetersToMiles(distance[0]),
+        );
+
+        const facilityCoordinatesWithDistances = [...facilityCoordinates].map(
+          (center, index) => ({
+            ...center,
+            distance: nearbyDistances[index],
+          }),
+        );
+
+        setNearbyVADistances(facilityCoordinatesWithDistances);
+      };
+
+      fetchDrivingData();
+      return false;
+    },
+    [props, originalCoordinates, nearbyVADistances],
+  );
+  const normalizeFetchedFacilityProperties = useCallback(
+    vc => {
+      let centerDistance = false;
+
+      if (isFinishedLoading(props) && nearbyVADistances.length) {
+        const facilityDistance = nearbyVADistances.find(
+          distance => distance.id === vc.id,
+        );
+        centerDistance = facilityDistance.distance;
+      }
+      return buildFacility(vc, centerDistance);
+    },
+    [nearbyVADistances, props],
+  );
+
+  const normalizeFetchedFacilities = vcs => {
+    return vcs
+      .map(vc => normalizeFetchedFacilityProperties(vc))
+      .sort((a, b) => a.distance - b.distance)
+      .reduce(
+        (acc, vaf) => {
+          if (vaf.source === 'Health' && acc[0] === null) {
+            acc[0] = vaf;
+          } else if (vaf.source === 'VetCenter' && acc[1] === null) {
+            acc[1] = vaf;
+          } else if (vaf.source === 'Cemetery' && acc[2] === null) {
+            acc[2] = vaf;
+          }
+          return acc;
```

Comment:
**Suggestion (non-blocking):** It looks like this matches the intention to have one of each type of center (Health, Vet Center, and Facility) in the Nearby VA Facilities. I got a little thrown by it being in a function for normalizing the fetched facilities. A comment above this `reduce` function may be helpful for reducing the cognitive load on our future selves.

Reply:
I will add a comment.